---
title: "Bacterial diversity in agricultural and prairie strip soils"
author: "Jared Flater"
date: "5/18/2020"
output:
  html_document:
    df_print: paged
    fig_width: 8.5
    fig_height: 11
    fig_caption: true
    toc: true
    toc_float: true
    number_sections: false
    code_folding: hide
---

```{r setup, include=FALSE}
library(phyloseq)
library(tidyverse)
library(ggplot2)
library(stats)
library(agricolae)
library(ggpubr)
library(kableExtra)
library(phylosmith)
library(data.table)
library(microbiome)
library(vegan)
library(gridExtra)
```

# Introduction

Diversity in microbial communities is a challenging question to tackle. Here we explore the alpha and beta diversity of the two rainfall simulations. We then use a phylogenetic tree to add another dimension to out alpha and beta diversity analysis. Using a phylogenetic tree allows us to incorporate evolutionary differences in sequences, which may be more informative only using number and abundance of species in a sample. 

We will look at 3 measures of diversity: observed number of species, Shannon diversity index, and Simpson diversity index. 

![Diversity Formulas and descriptions](images/diversityinfo.png)

```{r}
# Read in all strips samples
altstrips <- readRDS("data/all_samples_merged_STRIPS.RDS") 
head(taxa_names(altstrips))
taxa_names(altstrips) <- paste0("ASV", seq(ntaxa(altstrips)))
```

First we can check the number of reads in each sample, after evaluating a rarefaction curve, we found that ~10,000k reads sufficiently captured the species diversity in each sample. We can look at which samples are below 10k reads and that would be lost if we rarefy to 10k reads in each sample. 

```{r}
# simple plot of sample sums
data.frame(sample_sums(altstrips)) %>%
  cbind(., data.frame(sample_data(altstrips))) %>%
  rownames_to_column(var = "Sample") %>%
  as_tibble() %>%
  filter(sample_sums.altstrips. <= 10000) %T>% {print(paste0(nrow(.)," samples with less than 10k"))} %>% # print number of samples with less than 10k to the console
  ggplot(aes(x = reorder(Sample, sample_sums.altstrips.), y = sample_sums.altstrips.)) + 
  geom_bar(stat = "identity", aes(fill = matrix)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  facet_grid(depth~experiment)
```
```{r}
theme_my <- function(base_size = 7, base_family = "Palatino")
{
  txt <- element_text(size = 6, colour = "black", face = "plain")
  bold_txt <- element_text(size = 7, colour = "black", face = "bold")
  
  theme_bw(base_size = base_size, base_family = base_family) +
    theme(
      #legend.key = element_blank(), 
      #strip.background = element_blank(), 
      
      text = txt, 
      plot.title = txt, 
      
      axis.title = bold_txt, 
      axis.text = txt,
      axis.text.x = element_text(angle = 25, vjust = 1, hjust = 1),
      
      legend.title = bold_txt, 
      legend.text = txt) 
}
```

```{r rarefaction curve}

physeq <- altstrips

set.seed(42)

calculate_rarefaction_curves <- function(psdata, measures, depths, parallel=FALSE) {
  require('plyr') # ldply
  require('reshape2') # melt
  require('doParallel')
  
  # set parallel options if required
  if (parallel) {
    paropts  <- list(.packages=c("phyloseq", "reshape2"))
  } else {
    paropts  <- NULL
  }
  
  estimate_rarified_richness <- function(psdata, measures, depth) {
    if(max(sample_sums(psdata)) < depth) return()
    psdata <- prune_samples(sample_sums(psdata) >= depth, psdata)
    
    rarified_psdata <- rarefy_even_depth(psdata, depth, verbose = FALSE)
    
    alpha_diversity <- estimate_richness(rarified_psdata, measures = measures)
    
    # as.matrix forces the use of melt.array, which includes the Sample names (rownames)
    molten_alpha_diversity <- melt(as.matrix(alpha_diversity), varnames = c('Sample', 'Measure'), value.name = 'Alpha_diversity')
    
    molten_alpha_diversity
  }
  
  names(depths) <- depths # this enables automatic addition of the Depth to the output by ldply
  rarefaction_curve_data <- ldply(depths, estimate_rarified_richness, psdata = psdata, measures = measures, .id = 'Depth', .progress = ifelse(interactive() && ! parallel, 'text', 'none'), .parallel=parallel, .paropts=paropts)
  
  # convert Depth from factor to numeric
  rarefaction_curve_data$Depth <- as.numeric(levels(rarefaction_curve_data$Depth))[rarefaction_curve_data$Depth]
  
  rarefaction_curve_data
}

# Loading required library and displaying core configuration
pkgTest <- function(x) {
    if (!require(x,character.only = TRUE))
    {
     install.packages(x,dep=TRUE)
       if(!require(x,character.only = TRUE)) stop("Package not found")
    }
  }

pkgTest('doParallel')
library(doParallel)


#####
#Read in your physeq object, mine is inc.physeq
library(phyloseq)
library(tidyverse)



# Use multiple cores
detectCores(all.tests=TRUE)
# 48

# Setting up and registering the cluster
cl <- makeCluster(detectCores(all.tests=TRUE))
registerDoParallel(cl)

rarefaction_curve_data <- calculate_rarefaction_curves(physeq, c('Observed'), rep(c(1, 10, 100, 1000, 1:100 * 1000), each = 10))
#####

# Shut down cluster
stopCluster(cl)
rm(cl)
registerDoSEQ()

summary(rarefaction_curve_data)

rarefaction_curve_data_summary <- ddply(rarefaction_curve_data, c('Depth', 'Sample', 'Measure'), summarise, Alpha_diversity_mean = mean(Alpha_diversity), Alpha_diversity_sd = sd(Alpha_diversity))

sd <- as_tibble(sample_data(physeq)) %>%
  mutate_if(is.character, str_replace_all, pattern = '-', replacement = '.')

rarefaction_curve_data_summary_verbose <- merge(rarefaction_curve_data_summary, sd, by.x = 'Sample', by.y = 'unique_id')
library('ggplot2')

plot <- ggplot(
  data = rarefaction_curve_data_summary_verbose,
  mapping = aes(
    x = Depth,
    y = Alpha_diversity_mean,
    ymin = Alpha_diversity_mean - Alpha_diversity_sd,
    ymax = Alpha_diversity_mean + Alpha_diversity_sd,
    colour = treatment,
    group = Sample
  )
) + geom_line(
) + geom_pointrange(
) 
plot$layers[[2]] <- NULL
plot 
```

Let's start by looking at the Shannon and Simpson diversity index between crop and ag for baseline samples at WORLE and Armstrong, before rainfall simulation. This will include all treatments because manure had not been applied at this point, making it fair to compare all treatments without worrying about the influence of the manure community. 
```{r}
altstrips <- subset_samples(altstrips, !soil_type %in% "border" & matrix %in% "soil" & treatment %in% "no_manure_strip") %>%
  filter_taxa(function(x) sum(x) >= 2, T)
altstrips
max(sample_sums(altstrips))
min(sample_sums(altstrips))
nsamples(altstrips)
ntaxa(altstrips)
summarize_phyloseq(altstrips)

tdt <- sample_sums(altstrips) %>%
  enframe(name = "ID", value = "Sample_sum")

tdt <- as_tibble(sample_data(altstrips)) %>%
  cbind(., sample_sums(altstrips)) %>%
  dplyr::group_by(experiment, soil_type, sample_day, depth) %>%
  dplyr::summarize(Mean_sample_sums = mean(`sample_sums(altstrips)`, na.rm = TRUE), n_samples = n()) %>%
  write_csv("data/nSamplesStripsDiversity.csv")
```

```{r}
library(doParallel)
# Use multiple cores
detectCores(all.tests = TRUE)
# 6

# Setting up and registering the cluster
cl <- makeCluster(detectCores(all.tests = TRUE))
registerDoParallel(cl)

plot <- phylosmith::pcoa_phyloseq(altstrips, c('experiment', 'soil_type', 'depth'), circle = T)
#####

# Shut down cluster
stopCluster(cl)
rm(cl)
registerDoSEQ()
plot
ggsave("images/DiversityBetaWnAnD1nD2.png", plot = plot)
```

```{r}
Wdbase <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & sample_day %in% "Baseline" & !soil_type %in% "border" & matrix %in% "soil" & treatment %in% "no_manure_strip") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
Adbase <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & sample_day %in% "Baseline" & !soil_type %in% "border" & matrix %in% "soil" & treatment %in% "no_manure_strip") %>%
  filter_taxa(function(x) sum(x) >= 1, T)


W_Base_Shannon <- sample_data(Wdbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Wdbase, )) %>%
  ggboxplot(x = "soil_type", y = "Shannon", add = "jitter", notch = F, color = "soil_type", size = .2) + 
  stat_compare_means(size = 2) +
  ggtitle("Shannon diversity index of WOR baseline soils faceted by depth") +
  ylab("Shannon Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

print(min(taxa_sums(Wdbase)))

# Armstrong
A_Base_Shannon <- sample_data(Adbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Adbase, )) %>%
  ggboxplot(x = "soil_type", y = "Shannon", add = "jitter", notch = F, color = "soil_type", size = .2) + 
  stat_compare_means(size = 2) +
  ggtitle("Shannon diversity index of ARM baseline soils faceted by depth") +
  ylab("Shannon Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

print(min(taxa_sums(Wdbase)))
ggsave("images/WnA_Base_Shannon.png", arrangeGrob(W_Base_Shannon, A_Base_Shannon), units = "cm", height = 18, width = 8)
group <- A_Base_Shannon$data %>%
  dplyr::group_by(soil_type, depth) %>%
  summarise(mshan = mean(Shannon), mobs = mean(Observed), msimp = mean(Simpson))
group
```
```{r}
rarecurve(t(otu_table(Wdbase)), step = , cex = 0.5)
# Simpson diversity
# WORLE
W_Base_Simpson <- sample_data(Wdbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Wdbase, )) %>%
  ggboxplot(x = "soil_type", y = "Simpson", add = "jitter", notch = F, color = "soil_type", size = .2) + 
  stat_compare_means(size = 2) +
  ggtitle("Simpson diversity index of WOR baseline soils faceted by depth") +
  ylab("Simpson Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

print(min(taxa_sums(Wdbase)))

# Armstrong
A_Base_Simpson <- sample_data(Adbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Adbase, )) %>%
  ggboxplot(x = "soil_type", y = "Simpson", add = "jitter", notch = F, color = "soil_type", size = .2) + 
  stat_compare_means(size = 2) +
  ggtitle("Simpson diversity index of ARM baseline soils faceted by depth") +
  ylab("Simpson Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

print(min(taxa_sums(Wdbase)))
ggsave("images/WnA_Base_Simpson.png", arrangeGrob(W_Base_Simpson, A_Base_Simpson), units = "cm", height = 18, width = 8)
```
```{r}
W_Base_Observed <- sample_data(Wdbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Wdbase, )) %>%
  ggboxplot(x = "soil_type", y = "Observed", add = "jitter", notch = F, color = "soil_type", size = .2) + 
  stat_compare_means(size = 2) +
  ggtitle("Observed diversity index of WOR baseline soils faceted by depth") +
  ylab("Observed Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

print(min(taxa_sums(Wdbase)))

# Armstrong
A_Base_Observed <- sample_data(Adbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Adbase, )) %>%
  ggboxplot(x = "soil_type", y = "Observed", add = "jitter", notch = F, color = "soil_type", size = .2) + 
  stat_compare_means(size = 2) +
  ggtitle("Observed diversity index of ARM baseline soils faceted by depth") +
  ylab("Observed Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

print(min(taxa_sums(Wdbase)))
ggsave("images/WnA_Base_Observed.png", arrangeGrob(W_Base_Observed, A_Base_Observed), units = "cm", height = 18, width = 8)
```
# Across the plot variation

For the 6 plots that contain strips, we will examine the changes in diversity across space, s1 was the sample furthest into the crop and s9 is the sample furthest into the strip. We are only plotting no manured strip plots. 
```{r}
Wdbasestrip <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & !treatment %in% c("manured_control", "manured_strip") & matrix %in% "soil")
kable(data.frame(sample_data(Wdbasestrip))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
Wdbasestrip

data <- sample_data(Wdbasestrip) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Wdbasestrip, )) 
data %>%
  ggboxplot(x = "in_plot_location", y = "Observed", add = "jitter", notch = F, color = "soil_type") + 
  theme(legend.position = "none") +
  ggtitle("Observed diversity of Worle soils across space") +
  ylab("Observed Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_grid(depth ~ sample_day)

data %>%
  ggboxplot(x = "in_plot_location", y = "Shannon", add = "jitter", notch = F, color = "soil_type") + 
  theme(legend.position = "none") +
  ggtitle("Shannon diversity of Worle soils across space") +
  ylab("Shannon Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_grid(depth ~ sample_day)

data %>%
  ggboxplot(x = "in_plot_location", y = "Simpson", add = "jitter", notch = F, color = "soil_type") + 
  theme(legend.position = "none") +
  ggtitle("Simpson diversity of Worle soils across space") +
  ylab("Simpson Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_grid(depth ~ sample_day)
```
```{r}
Adbasestrip <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & !treatment %in% c("manured_control", "manured_strip") & matrix %in% "soil")

kable(data.frame(sample_data(Wdbasestrip))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
Adbasestrip


data <- sample_data(Adbasestrip) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Adbasestrip, )) 
data %>%
  ggboxplot(x = "in_plot_location", y = "Observed", add = "jitter", notch = F, color = "soil_type") + 
  theme(legend.position = "none") +
  ggtitle("Observed diversity of Armstrong soils across space") +
  ylab("Observed Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_grid(depth ~ sample_day)

data %>%
  ggboxplot(x = "in_plot_location", y = "Shannon", add = "jitter", notch = F, color = "soil_type") + 
  theme(legend.position = "none") +
  ggtitle("Shannon diversity of Armstrong soils across space") +
  ylab("Shannon Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_grid(depth ~ sample_day)

data %>%
  ggboxplot(x = "in_plot_location", y = "Simpson", add = "jitter", notch = F, color = "soil_type") + 
  theme(legend.position = "none") +
  ggtitle("Simpson diversity of Armstrong soils across space") +
  ylab("Simpson Diversity") +
  scale_color_viridis_d() +
  scale_x_discrete(name = "Soil type", labels = c("crop" = "Ag soil", "strip" = "Prairie strip soil")) +
  facet_grid(depth ~ sample_day)
```
Prior to rainfall simulation, there were no differences in Shannon or Simpson or Richness (observed) diversity of the baseline soils when comparing prairie soil to crop soil at each site (depth 1). 

Let's now look at differences by time for each site. 

```{r Over Time Shannon}
# Subset to WOR and ARM depth one no manure, exclude the border sample from each plot. 
Wd1 <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & treatment %in% "no_manure_strip" & !soil_type %in% "border" & matrix %in% "soil")
Ad1 <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & treatment %in% "no_manure_strip" & !soil_type %in% "border" & matrix %in% "soil")
```
```{r}
kable(data.frame(sample_data(Wd1))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
Wsample_data <- sample_data(Wd1) %>%
  lapply(., factor) %>%
  as_tibble() 

Wrichness <- estimate_richness(Wd1) 

Wsample_data <- cbind(Wsample_data, Wrichness)

wp <- ggboxplot(Wsample_data, x = "sample_day", y = "Shannon", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

wplot <- wp + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WOR soils") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
wplot
ggsave("images/WorleD1Shannon.png", plot = wplot, device = "png", width = 6, height = 4, units = "in")

wp <- ggboxplot(Wsample_data, x = "sample_day", y = "Observed", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

wplot <- wp + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Observed diversity index of WOR soils") +
  ylab("Observed Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
wplot
ggsave("images/WorleD1Observed.png", plot = wplot, device = "png", width = 6, height = 4, units = "in")

wp <- ggboxplot(Wsample_data, x = "sample_day", y = "Simpson", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

wplot <- wp + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Simpson diversity index of WOR soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
wplot
ggsave("images/WorleD1Simpson.png", plot = wplot, device = "png", width = 6, height = 4, units = "in")
```
```{r}
kable(data.frame(sample_data(Ad1))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
Asample_data <- sample_data(Ad1) %>%
  lapply(., factor) %>%
  as_tibble() 

Arichness <- estimate_richness(Ad1) 

Asample_data <- cbind(Asample_data, Arichness)

ap <- ggboxplot(Asample_data, x = "sample_day", y = "Shannon", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

aplot <- ap + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of ARM soils") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
aplot
ggsave("images/ARMD1Shannon.png", plot = aplot, device = "png", width = 6, height = 4, units = "in")

ap <- ggboxplot(Asample_data, x = "sample_day", y = "Observed", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

aplot <- ap + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Observed diversity index of ARM soils") +
  ylab("Observed Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
aplot
ggsave("images/ARMD1Observed.png", plot = aplot, device = "png", width = 6, height = 4, units = "in")

ap <- ggboxplot(Asample_data, x = "sample_day", y = "Simpson", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")

aplot <- ap + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Simpson diversity index of ARM soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
aplot
ggsave("images/ARMD1Simpson.png", plot = aplot, device = "png", width = 6, height = 4, units = "in")
```


We see some differences on T002, which was two days after the rainfall simulation. The added moisture likely caused an increase in bacteria. Had we measured microbial biomass carbon, I would have expected to see an increase. We could confirm the increase by using 16s primers and qPCR for these samples. 
```{r}
my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))
Sp <- Wsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE prairie soils") +
  ylab("Shannon Diversity") +
  theme(axis.title.x=element_blank())
Sp

Op <- Wsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Observed", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Observed diversity index of WORLE prairie soils") +
  ylab("Observed Diversity") +
  theme(axis.title.x=element_blank())
Op

SMp <- Wsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Simpson", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Simpson diversity index of WORLE prairie soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day")
SMp
ggsave("images/WORoverTimpPraire.png", arrangeGrob(Op, Sp, SMp), units = "cm", height = 25, width = 10)
```
```{r}
Sp <- Wsample_data %>%
  filter(soil_type == "crop") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE crop soils") +
  ylab("Shannon Diversity") +
  theme(axis.title.x=element_blank())
Sp

Op <- Wsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Observed", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Observed diversity index of WORLE crop soils") +
  ylab("Observed Diversity") +
  theme(axis.title.x=element_blank())
Op

SMp <- Wsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Simpson", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Simpson diversity index of WORLE crop soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day")
SMp
ggsave("images/WORoverTimpCrop.png", arrangeGrob(Op, Sp, SMp), units = "cm", height = 25, width = 10)
```
```{r}
my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T153"))
Sp <- Asample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of ARM prairie soils") +
  ylab("Shannon Diversity") +
  theme(axis.title.x=element_blank())
Sp

Op <- Asample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Observed", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Observed diversity index of ARM prairie soils") +
  ylab("Observed Diversity") +
  theme(axis.title.x=element_blank())
Op

SMp <- Asample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Simpson", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Simpson diversity index of ARM prairie soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day")
SMp
ggsave("images/ARMoverTimpPraire.png", arrangeGrob(Op, Sp, SMp), units = "cm", height = 25, width = 10)
```
```{r}
Sp <- Asample_data %>%
  filter(soil_type == "crop") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of ARM crop soils") +
  ylab("Shannon Diversity") +
  theme(axis.title.x=element_blank())
Sp

Op <- Asample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Observed", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Observed diversity index of ARM crop soils") +
  ylab("Observed Diversity") +
  theme(axis.title.x=element_blank())
Op

SMp <- Asample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Simpson", add = "jitter") +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none") + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Simpson diversity index of ARM crop soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day")
SMp
ggsave("images/ARMoverTimecrop.png", arrangeGrob(Op, Sp, SMp), units = "cm", height = 25, width = 10)
```


```{r Simpson}
p <- ggboxplot(Wsample_data, x = "sample_day", y = "Simpson", color = "soil_type", add = "jitter", notch = F) +
  facet_wrap( ~ depth) +
  theme_my() +
  theme(legend.position = "none")
plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Simpson diversity index of WOR detph 1 soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d()
plot
ggsave("images/WorleD1Simpson.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```
```{r Observed}
p <- ggboxplot(sample_data, x = "sample_day", y = "Observed", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Observed number of species in WORLE detph 1 soils") +
  ylab("Observed species") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1observed.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```
Let's take a look at Armstrong depth 1 soils: 

```{r armstrongdepth 1 subset}
# Subset to WORLE depth one no manure, exclude the border sample from each plot. 
Ad1 <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "1") 

kable(data.frame(sample_data(Ad1))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
```

```{r}
sample_data <- sample_data(Ad1) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Ad1, )) 

p <- ggboxplot(sample_data, x = "sample_day", y = "Observed", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Observed number of species in Armstrong detph 1 soils") +
  ylab("Observed species") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/ArmstrongD1observed.png", plot = plot, device = "png", width = 6, height = 4, units = "in")

```
```{r}
p <- ggboxplot(sample_data, x = "sample_day", y = "Shannon", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon number of species in Armstrong detph 1 soils") +
  ylab("Shannon species") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/ArmstrongD1Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```
```{r}
p <- ggboxplot(sample_data, x = "sample_day", y = "Simpson", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Simpson number of species in Armstrong detph 1 soils") +
  ylab("Simpson species") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/ArmstrongD1Simpson.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```



Rarefied Diversity

Rarefaction is important for making sound judgments when comparing two sites (samples) in this case, our sequencing wasn't even between all samples, so we are randomly re-sampling each at 10k reads. This removes a few samples that had less than 10k reads, we cannot include them in the comparison because the were under-sampled. 

![](images/Wd1RareCurveMinTax1.png)
```{r}
rWd1 <- rarefy_even_depth(Wd1, sample.size = 10000, rngseed = 12121212, trimOTUs = T)
```

```{r Rare Richness}
rsample_data1 <- sample_data(rWd1) %>%
  lapply(., factor) %>%
as_tibble()
richness <- estimate_richness(rWd1) 
rsample_data <- cbind(rsample_data1, richness)
rm(rsample_data1)
```

```{r Shannon rarefied}
p <- ggboxplot(rsample_data, x = "sample_day", y = "Shannon", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 soils, rarefied to 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/rWd1Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

# Prairie and Crop Pairwise

Just prairie and crop soils pairwise comparison, we can see if any of the days were significantly more or less diverse than the other. The lines on the graph connect the two samples that are being compared. 

Here we can see that there is more within season change of diversity in prairie soils as compared to ag soils. 

Sample day 2 and 21 have significant increases in diversity.
```{r}
p <- rsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 prairie soils, rarefied 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot

p <- rsample_data %>%
  filter(soil_type == "crop") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 ag soils, rarefied 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot
```

```{r}
p <- sample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "ACE", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("ACE diversity index of WORLE depth 1 prairie soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") 
plot
```

#### Depth 2 

What about the second depth of the soil samples, 7-15 cm? 

```{r depth 2 subset}
Wd2 <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "2") 
Wd2
kable(data.frame(sample_data(Wd2))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
```

```{r ACE 2}
sample_data <- sample_data(Wd2) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd2) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "ACE", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of WORLE detph 2 soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD2ACE.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

```{r Chao1 2}
sample_data <- sample_data(Wd2) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd2) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "Chao1", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Chao1 diversity index of WORLE detph 2 soils") +
  ylab("Chao1 Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD2Chao1.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

```{r}
rWd2 <- rarefy_even_depth(Wd2, sample.size = 10000, rngseed = 12121212, trimOTUs = T)
```

```{r Shannon rare 2}
sample_data2 <- sample_data(rWd2) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(rWd2) 
sample_data2 <- cbind(sample_data2, richness)

p <- ggboxplot(sample_data2, x = "sample_day", y = "Shannon", color = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 2 soils rarefied to 10k reads") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorlerD2Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```


```{r}
# sample data 2 is rarefied, this is shannon
p <- sample_data2 %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 2 Prairie soils, rarefied to 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot

p <- sample_data2 %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Simpson", add = "jitter")

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Simpson diversity index of WORLE detph 2 Prairie soils, rarefied to 10k") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day") 
plot
```
```{r}
p <- sample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "ACE", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("ACE diversity index of WORLE depth 2 Prairie soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") 
plot
```

### Armstrong Depth 1

Depth 1 over time, comparison between prairie and crop at each time
```{r}
Ad1 <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
Ad1
kable(data.frame(sample_data(Ad1))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
```

Baseline of all treatments depth 1 in Armstrong soils
```{r}
Abase <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & sample_day %in% "Baseline" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x) >= 1, T) 

sample_data(Abase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Abase)) %>%
  ggboxplot(x = "sample_day", y = "Simpson", color = "soil_type", add = "jitter", notch = T) + 
  stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Simpson diversity index of Armstrong baseline detph 1 soils") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day") + scale_color_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
```

# Beta Diversity

We will prune a lot of OTUs to make analysis quicker, observed at least 3 times in 20% of samples

```{r}
strips <- readRDS("data/all_samples_merged_STRIPS.RDS") 

Allbase <- subset_samples(strips, sample_day %in% "Baseline" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x >= 3) > (0.2*length(x)), T) 

rm(strips)
taxa_names(Allbase) <- paste0("ASV", seq(ntaxa(Allbase)))
```
```{r}
ordinate(Allbase, "PCoA", "bray") %>% 
  plot_ordination(Allbase, ., color = "soil_type", shape = "experiment", title = "Bray-Curtis PCoA") 
```
```{r}
ordinate(Allbase, "NMDS", "bray") %>% 
  plot_ordination(Allbase, ., color = "soil_type", shape = "experiment", title = "Bray-Curtis NMDS") 
```

# Phylogenetic tree 

Construct phylogenetic tree, from: https://f1000research.com/articles/5-1492/v2

```{r}
optimiezedTree <- readRDS("data/d1mintax5tree.RDS") 

Allbasewtree <- merge_phyloseq(Allbase, phy_tree(optimiezedTree))
Allbasewtree
```

```{r}
ordinate(Allbasewtree, "NMDS", "wunifrac") %>% 
  plot_ordination(Allbasewtree, ., color = "soil_type", shape = "experiment", title = "wunifrac") 
```
What about at just one site? WORLE:

```{r}
worle <- subset_samples(Allbasewtree, experiment %in% "Worle_Rainfall") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
worle
```
```{r}
ordinate(worle, "NMDS", "wunifrac") %>% 
  plot_ordination(worle, ., color = "soil_type", title = "wunifrac at WORLE") + 
  stat_ellipse(geom = "polygon", type = "norm", alpha = 0.1, aes(fill = soil_type)) 
adonis(phyloseq::distance(worle, method = "wunifrac") ~ soil_type, data = data.frame(sample_data(worle)))
```

```{r}
Wd1 <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "1")
head(taxa_names(Wd1))

singles <- function(phy){
   df <- as(otu_table(phy), "matrix") %>%
    as_tibble(rownames = "ASV") %>%
    filter_all(any_vars(. == 1)) %>%
    select_if(~max(.) > 0)
  return(df)
}
singles(Wd1)
```
```{r}
# install.packages("picante")
# library(picante)
# if( "FaithPD" %in% measures){
#   	outlist <- c(outlist, list(FaithPD = t(picante::pd(samp = OTU, tree = phy_tree(physeq), include.root = F))[1,] ))
# }
# faithd <- picante::pd(samp = OTU, tree = phy_tree(physeq), include.root = F)
```

