---
title: "Bacterial diversity in agricultural and prairie strip soils"
author: "Jared Flater"
date: "5/18/2020"
output:
  html_document:
    df_print: paged
    fig_width: 8.5
    fig_height: 11
    fig_caption: true
    toc: true
    toc_float: true
    number_sections: false
    code_folding: hide
---

```{r setup, include=FALSE}
library(phyloseq)
library(tidyverse)
library(ggplot2)
library(stats)
library(agricolae)
library(ggpubr)
```

Alpha diversity is the description of the number of species and sometimes the counts within species at different sites. 

Shannon diversity:     

ACE diversity:   

Chao1:    


```{r Load STRIPS, echo=F}
StripsPhy <- readRDS("data/merged_STRIPs.RDS")

StripsPhy <- StripsPhy %>%
  filter_taxa(function(x) sum(x) >= 1, T)
```

# Chao1 and ACE    

Both are nonparametric diversity estimates

```{r both depths}
plot <- plot_richness(StripsPhy, x = "soil_type", measures = c("Chao1", "ACE")) +
  geom_boxplot() +
  facet_grid(variable ~ sample_day)

plot
```

# Depth 1 WORLE

The first 7 cm of soil are depth one, we will compare these samples from prairie and ag portions of the plot

```{r depth 1 subset}
Wd1 <- subset_samples(StripsPhy, experiment %in% "Worle_Rainfall" & depth %in% 1 & !soil_type %in% "border") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
Wd1
```

Using phyloseq and it's built in `plot_richness`

```{r phylsoeq plot richness}
plot <- plot_richness(Wd1, x = "soil_type", measures = c("Chao1", "ACE")) +
  geom_boxplot()
plot
plot <- plot_richness(Wd1, x = "sample_day", measures = c("Chao1", "ACE"), color = "soil_type") +
  geom_boxplot()
plot
```
```{r stats for WORLE detph 1}
Wd1.Richness <- estimate_richness(Wd1, measures = c("Chao1", "ACE"))
ttest <- t(sapply(Wd1.Richness, function(x) unlist(t.test(x~sample_data(Wd1)$soil_type)[c("estimate","p.value","statistic","conf.int")])))
ttest
```

```{r}
physeq <- Wd1

c1 <- c("soil_type")
c2 <- c("sample_day")

sample_data <- sample_data(physeq) %>%
as_tibble()
sample_data$alpha <- estimate_richness(physeq, measures = c("ACE")) %>% 
  select(ACE) %>%
  unlist()
# Do ANOVA
sample_data <- sample_data %>%
  mutate(grouping = paste(sample_day, soil_type,  sep = 'x'))
# needed for how `aov` works
anova_result <- aov(alpha ~ grouping, sample_data)
anova_result
# Do Tukey's HSD test
tukey_result <- HSD.test(anova_result, "grouping", group = TRUE)
# Plot result
group_data <- tukey_result$groups[order(rownames(tukey_result$groups)),]
my_plot <- ggplot(sample_data, aes(x = sample_day, y = alpha)) +
  # geom_text(data = data.frame(),
  #           aes(x = rownames(group_data),
  #               y = max(sample_data$alpha) + 1,
  #               label = group_data$groups),
  #           col = 'black',
  #           size = 10) +
  geom_boxplot(aes(color = soil_type), position = "dodge") +
  ggtitle("Alpha diversity") +
  ylab("Alpha diversity index")
my_plot
```

# ggpubr 

ggpubr makes stats a lot easier

```{r ACE}
library(ggpubr)
sample_data <- sample_data(Wd1) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd1) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "ACE", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of WORLE detph 1 soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1ACE.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

We see some differences on T002, which was two days after the rainfall simulation. The added moisture likely caused an increase in bacteria. Had we measured microbial biomass carbon, I would have expected to see an increase. We could confirm the increase by using 16s primers and qPCR for these samples. 

```{r Chao1}
library(ggpubr)
sample_data <- sample_data(Wd1) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd1) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "Chao1", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Chao1 diversity index of WORLE detph 1 soils") +
  ylab("Chao1 Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1Chao1.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```
```{r Shannon}
library(ggpubr)
sample_data <- sample_data(Wd1) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd1) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "Shannon", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 soils") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```
```{r}
rWd1 <- rarefy_even_depth(Wd1, sample.size = 10000, rngseed = 12121212, trimOTUs = T)
```

Rarefied Diversity

Rarefaction is important for making sound judgements when comparing two sites (samples) in this case, our sequencing wasn't even between all samples, so we are randomly re-sampling each at 10k reads. This removes a few samples that had less than 10k reads, we cannot include them in the comparison becuase the were undersampled. Yet, our conclusions remain the same (T002 differences), with the exception of an additional difference detected on day 21 for ACE and Chao1.

```{r Rare Richness}
library(ggpubr)
sample_data <- sample_data(rWd1) %>%
  lapply(., factor) %>%
as_tibble()
richness <- estimate_richness(rWd1) 
sample_data <- cbind(sample_data, richness)
colnames(sample_data)
```
```{r ACE rarefied}
p <- ggboxplot(sample_data, x = "sample_day", y = "ACE", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of WORLE detph 1 soils, rarefied to 10k") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/rWd1ACE.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

```{r Chao1 rarefied}
p <- ggboxplot(sample_data, x = "sample_day", y = "Chao1", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Chao1 diversity index of WORLE detph 1 soils, rarefied to 10k") +
  ylab("Chao1 Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/rWd1Chao1.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

```{r Shannon rarefied}
p <- ggboxplot(sample_data, x = "sample_day", y = "Shannon", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 soils, rarefied to 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/rWd1Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

# Prairie Pairwise

Just prairie soils pairwise comparison, we can see if any of the days were signigicantly more or less diverse than the other. The lines on the graph conncet the two samples that are being compared. 

```{r}
levels(as.factor(sample_data$sample_day))
levels(as.factor(p$sample_day))
p <- sample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")
p
my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 Prairie soils, rarefied to 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot
```
```{r}
levels(as.factor(sample_data$sample_day))
levels(as.factor(p$sample_day))
p <- sample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "ACE", add = "jitter")
p
my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("ACE diversity index of WORLE depth 1 Prairie soils, rarefied to 10k") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") 
plot
```

Trying a violin plot

```{r}
p <- sample_data %>%
  filter(soil_type == "crop") %>%
  ggviolin(x = "sample_day", y = "ACE", fill = "sample_day",
         add = "boxplot", add.params = list(fill = "gray"))
p
my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("ACE diversity index of WORLE depth 1 Crop soils, rarefied to 10k") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + 
  scale_fill_viridis_d() + 
  theme(legend.position = "none") 
plot
```

