---
title: "Bacterial diversity in agricultural and prairie strip soils"
author: "Jared Flater"
date: "5/18/2020"
output:
  html_document:
    df_print: paged
    fig_width: 8.5
    fig_height: 11
    fig_caption: true
    toc: true
    toc_float: true
    number_sections: false
    code_folding: hide
---

```{r setup, include=FALSE}
library(phyloseq)
library(tidyverse)
library(ggplot2)
library(stats)
library(agricolae)
library(ggpubr)
library(kableExtra)
library(phylosmith)
```

Alpha diversity is the description of the number of species and sometimes the counts within species at different sites. 

Shannon diversity:     

ACE diversity:   

Chao1:    

```{r}
altstrips <- readRDS("data/all_samples_merged_STRIPS.RDS") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
altstrips
```
```{r}
data.frame(sample_sums(altstrips)) %>%
  cbind(., data.frame(sample_data(altstrips))) %>%
  rownames_to_column(var = "Sample") %>%
  as_tibble() %>%
  filter(sample_sums.altstrips. <= 10000) %T>% {print(paste0(nrow(.)," samples with less than 10k"))} %>%
  ggplot(aes(x = reorder(Sample, sample_sums.altstrips.), y = sample_sums.altstrips.)) + 
  geom_bar(stat = "identity", aes(fill = matrix)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  facet_grid(depth~experiment)
```

## Chao1 and ACE and Shannon   

### Depth 1 WORLE

The first 7 cm of soil are depth one, we will compare these samples from prairie and ag portions of the plot.

```{r depth 1 subset}
Wd1 <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
Wd1
kable(data.frame(sample_data(Wd1))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
```

# ggpubr 

ggpubr makes stats a lot easier

Let's start by looking at the ACE diversity between crop and ag for baseline samples, before rainfall simulation. This will include all treatments.
```{r}
Wdbase <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & sample_day %in% "Baseline" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x) >= 1, T) 

sample_data(Wdbase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Wdbase)) %>%
  ggboxplot(x = "sample_day", y = "ACE", fill = "soil_type", add = "jitter", notch = T) + 
  stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of WORLE baseline detph 1 soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
```

```{r ACE}
library(ggpubr)
sample_data <- sample_data(Wd1) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd1) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "ACE", fill = "soil_type", add = "jitter", notch = F)

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of WORLE detph 1 soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1ACE.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

We see some differences on T002, which was two days after the rainfall simulation. The added moisture likely caused an increase in bacteria. Had we measured microbial biomass carbon, I would have expected to see an increase. We could confirm the increase by using 16s primers and qPCR for these samples. 


Chao1 looks very similar
```{r Chao1}
p <- ggboxplot(sample_data, x = "sample_day", y = "Chao1", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Chao1 diversity index of WORLE detph 1 soils") +
  ylab("Chao1 Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1Chao1.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```
```{r Shannon}
p <- ggboxplot(sample_data, x = "sample_day", y = "Shannon", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 soils") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD1Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

Rarefied Diversity

Rarefaction is important for making sound judgements when comparing two sites (samples) in this case, our sequencing wasn't even between all samples, so we are randomly re-sampling each at 10k reads. This removes a few samples that had less than 10k reads, we cannot include them in the comparison becuase the were undersampled. 

![](images/Wd1RareCurveMinTax1.png)
```{r}
rWd1 <- rarefy_even_depth(Wd1, sample.size = 10000, rngseed = 12121212, trimOTUs = T)
```

```{r Rare Richness}
rsample_data1 <- sample_data(rWd1) %>%
  lapply(., factor) %>%
as_tibble()
richness <- estimate_richness(rWd1) 
rsample_data <- cbind(rsample_data1, richness)
rm(rsample_data1)
```

```{r Shannon rarefied}
p <- ggboxplot(rsample_data, x = "sample_day", y = "Shannon", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 soils, rarefied to 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/rWd1Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

# Prairie and Crop Pairwise

Just prairie and crop soils pairwise comparison, we can see if any of the days were signigicantly more or less diverse than the other. The lines on the graph conncet the two samples that are being compared. 

Here we can see that there is more within season change of diversity in prairie soils as compared to ag soils. 

Sample day 2 and 21 have significant increases in diversity.
```{r}
p <- rsample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 prairie soils, rarefied 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot

p <- rsample_data %>%
  filter(soil_type == "crop") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 1 ag soils, rarefied 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot
```

```{r}
p <- sample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "ACE", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("ACE diversity index of WORLE depth 1 prairie soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") 
plot
```

#### Depth 2 

What about the second depth of the soil samples, 7-15 cm? 

```{r depth 2 subset}
Wd2 <- subset_samples(altstrips, experiment %in% "Worle_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "2") %>%
  filter_taxa(function(x) sum(x) >= 2, T)
Wd2
kable(data.frame(sample_data(Wd2))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
```

```{r ACE 2}
sample_data <- sample_data(Wd2) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd2) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "ACE", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of WORLE detph 2 soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD2ACE.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

```{r Chao1 2}
sample_data <- sample_data(Wd2) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(Wd2) 
sample_data <- cbind(sample_data, richness)

p <- ggboxplot(sample_data, x = "sample_day", y = "Chao1", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Chao1 diversity index of WORLE detph 2 soils") +
  ylab("Chao1 Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorleD2Chao1.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```

```{r}
rWd2 <- rarefy_even_depth(Wd2, sample.size = 10000, rngseed = 12121212, trimOTUs = T)
```

```{r Shannon rare 2}
sample_data2 <- sample_data(rWd2) %>%
  lapply(., factor) %>%
  as_tibble() 
  
richness <- estimate_richness(rWd2) 
sample_data2 <- cbind(sample_data2, richness)

p <- ggboxplot(sample_data2, x = "sample_day", y = "Shannon", fill = "soil_type", add = "jitter")

plot <- p + stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 2 soils rarefied to 10k reads") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
plot
ggsave("images/WorlerD2Shannon.png", plot = plot, device = "png", width = 6, height = 4, units = "in")
```


```{r}
# sample data 2 is rarefied, this is shannon
p <- sample_data2 %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Shannon", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Shannon diversity index of WORLE detph 2 Prairie soils, rarefied to 10k") +
  ylab("Shannon Diversity") +
  xlab("Sampling Day") 
plot

p <- sample_data2 %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "Simpson", add = "jitter")

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("Simpson diversity index of WORLE detph 2 Prairie soils, rarefied to 10k") +
  ylab("Simpson Diversity") +
  xlab("Sampling Day") 
plot
```
```{r}
p <- sample_data %>%
  filter(soil_type == "strip") %>%
  ggboxplot(x = "sample_day", y = "ACE", add = "jitter")

my_comparisons <- list( c("T002", "T014"), c("T002", "T000"), c("Baseline", "T000"), c("T014", "T021"), c("T021", "T042"))

plot <- p + stat_compare_means(comparisons = my_comparisons, label = "p.signif") +
  ggtitle("ACE diversity index of WORLE depth 2 Prairie soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") 
plot
```

### Armstrong Depth 1

Depth 1 over time, comparison between prairie and crop at each time
```{r}
Ad1 <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & manure_treatment %in% "N" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x) >= 1, T)
Ad1
kable(data.frame(sample_data(Ad1))) %>%
  kable_styling(bootstrap_options = "striped") %>%
  scroll_box(width = "100%", height = "600px")
```

Baseline of all treatments depth 1 in Armstrong soils
```{r}
Abase <- subset_samples(altstrips, experiment %in% "Armstrong_Rainfall" & sample_day %in% "Baseline" & !soil_type %in% "border" & depth %in% "1") %>%
  filter_taxa(function(x) sum(x) >= 1, T) 

sample_data(Abase) %>%
  lapply(., factor) %>%
  as_tibble() %>%
  cbind(., estimate_richness(Abase)) %>%
  ggboxplot(x = "sample_day", y = "ACE", fill = "soil_type", add = "jitter", notch = T) + 
  stat_compare_means(aes(group = soil_type), label = "p.signif") +
  ggtitle("ACE diversity index of Armstrong baseline detph 1 soils") +
  ylab("ACE Diversity") +
  xlab("Sampling Day") + scale_fill_viridis_d(name = "Managment", labels = c("Ag", "Prairie"))
```