<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jared Flater" />


<title>Bacterial diversity in agricultural and prairie strip soils</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/lumen.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 54px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h2 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h3 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h4 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h5 {
  padding-top: 59px;
  margin-top: -59px;
}
.section h6 {
  padding-top: 59px;
  margin-top: -59px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Strips soil bacteria analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Diversity_Stats.html">Alph Diversity</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Bacterial diversity in agricultural and prairie strip soils</h1>
<h4 class="author">Jared Flater</h4>
<h4 class="date">5/18/2020</h4>

</div>


<pre class="r"><code>data(&quot;soilrep&quot;)
soilrep</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 16825 taxa and 56 samples ]
## sample_data() Sample Data:       [ 56 samples by 4 sample variables ]</code></pre>
<pre class="r"><code>erich &lt;- estimate_richness(soilrep, measures = c(&quot;Observed&quot;, &quot;Shannon&quot;, &quot;Simpson&quot;, &quot;Fisher&quot;))
ttest &lt;- t(sapply(erich, function(x) unlist(t.test(x~sample_data(soilrep)$warmed)[c(&quot;estimate&quot;,&quot;p.value&quot;,&quot;statistic&quot;,&quot;conf.int&quot;)])))
ttest</code></pre>
<pre><code>##          estimate.mean in group no estimate.mean in group yes   p.value
## Observed              1133.1428571                1109.392857 0.8222405
## Shannon                  6.7239381                   6.725982 0.9776912
## Simpson                  0.9980636                   0.998125 0.6300884
## Fisher                1400.0104151                1422.904606 0.7783465
##          statistic.t     conf.int1    conf.int2
## Observed  0.22579676 -1.872919e+02 2.347919e+02
## Shannon  -0.02809354 -1.479344e-01 1.438458e-01
## Simpson  -0.48438244 -3.156804e-04 1.928421e-04
## Fisher   -0.28289961 -1.851811e+02 1.393927e+02</code></pre>
<pre class="r"><code>StripsPhy &lt;- readRDS(&quot;data/merged_STRIPs.RDS&quot;)

StripsPhy &lt;- StripsPhy %&gt;%
  filter_taxa(function(x) sum(x) &gt;= 1, T)

StripsPhy</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 48381 taxa and 586 samples ]
## sample_data() Sample Data:       [ 586 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 48381 taxa by 6 taxonomic ranks ]
## refseq()      DNAStringSet:      [ 48381 reference sequences ]</code></pre>
<pre class="r"><code>min(taxa_sums(StripsPhy))</code></pre>
<pre><code>## [1] 1</code></pre>
<pre class="r"><code>head(StripsPhy@sam_data$matrix)</code></pre>
<pre><code>## [1] &quot;soil&quot; &quot;soil&quot; &quot;soil&quot; &quot;soil&quot; &quot;soil&quot; &quot;soil&quot;</code></pre>
<div id="chao1-and-ace" class="section level1">
<h1>Chao1 and ACE</h1>
<p>Both are nonparametric diversity estimates</p>
<pre class="r"><code>plot &lt;- plot_richness(StripsPhy, x = &quot;soil_type&quot;, measures = c(&quot;Chao1&quot;, &quot;ACE&quot;)) +
  geom_boxplot() +
  facet_grid(variable ~ sample_day)

plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/both%20depths-1.png" width="816" /></p>
</div>
<div id="depth-1-worle" class="section level1">
<h1>Depth 1 WORLE</h1>
<p>The first 7 cm of soil are depth one, we will compare these samples from prairie and ag portions of the plot</p>
<pre class="r"><code>Wd1 &lt;- subset_samples(StripsPhy, experiment %in% &quot;Worle_Rainfall&quot; &amp; depth %in% 1 &amp; !soil_type %in% &quot;border&quot;) %&gt;%
  filter_taxa(function(x) sum(x) &gt;= 1, T)
Wd1</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 15725 taxa and 138 samples ]
## sample_data() Sample Data:       [ 138 samples by 12 sample variables ]
## tax_table()   Taxonomy Table:    [ 15725 taxa by 6 taxonomic ranks ]
## refseq()      DNAStringSet:      [ 15725 reference sequences ]</code></pre>
<p>Using phyloseq and it’s built in <code>plot_richness</code></p>
<pre class="r"><code>plot &lt;- plot_richness(Wd1, x = &quot;soil_type&quot;, measures = c(&quot;Chao1&quot;, &quot;ACE&quot;)) +
  geom_boxplot()
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/phylsoeq%20plot%20richness-1.png" width="816" /></p>
<pre class="r"><code>plot &lt;- plot_richness(Wd1, x = &quot;sample_day&quot;, measures = c(&quot;Chao1&quot;, &quot;ACE&quot;), color = &quot;soil_type&quot;) +
  geom_boxplot()
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/phylsoeq%20plot%20richness-2.png" width="816" /></p>
<pre class="r"><code>Wd1.Richness &lt;- estimate_richness(Wd1, measures = c(&quot;Chao1&quot;, &quot;ACE&quot;))
ttest &lt;- t(sapply(Wd1.Richness, function(x) unlist(t.test(x~sample_data(Wd1)$soil_type)[c(&quot;estimate&quot;,&quot;p.value&quot;,&quot;statistic&quot;,&quot;conf.int&quot;)])))
ttest</code></pre>
<pre><code>##          estimate.mean in group crop estimate.mean in group strip
## Chao1                   7.583623e+02                 8.546667e+02
## se.chao1                1.724528e-04                 1.905859e-04
## ACE                     7.583652e+02                 8.546697e+02
## se.ACE                  1.305958e+01                 1.361247e+01
##             p.value statistic.t     conf.int1   conf.int2
## Chao1    0.05784381  -1.9150152 -1.958618e+02 3.253087941
## se.chao1 0.94386121  -0.0705492 -5.264651e-04 0.000490199
## ACE      0.05784510  -1.9150050 -1.958624e+02 3.253609303
## se.ACE   0.12729459  -1.5346724 -1.265635e+00 0.159853277</code></pre>
<p><code>r  # compare_alpha &lt;- function(var1, physeq) {  #   # Calcuate alpha diversity  #   sample_data &lt;- sample_data(physeq) %&gt;%  #   as_tibble()  #   sample_data$alpha &lt;- estimate_richness(physeq, measures = c(&quot;ACE&quot;)) %&gt;%   #     select(ACE) %&gt;%  #     unlist()  #   # Do ANOVA  #   sample_data$grouping &lt;- sample_data[[grouping_var]] # needed for how `aov` works  #   anova_result &lt;- aov(alpha ~ grouping, sample_data)  #   # Do Tukey's HSD test  #   tukey_result &lt;- HSD.test(anova_result, &quot;grouping&quot;, group = TRUE)  #   # Plot result  #   group_data &lt;- tukey_result$groups[order(rownames(tukey_result$groups)),]  #   my_plot &lt;- ggplot(sample_data, aes(x = grouping, y = alpha)) +  #     geom_text(data = data.frame(),  #               aes(x = rownames(group_data),  #                   y = max(sample_data$alpha) + 1,  #                   label = group_data$groups),  #               col = 'black',  #               size = 10) +  #     geom_boxplot() +  #     ggtitle(&quot;Alpha diversity&quot;) +  #     xlab(grouping) +  #     ylab(&quot;Alpha diversity index&quot;)  #   # Return plot  #   return(my_plot)  # }  #</code></p>
<p><code>r  # compare_alpha(&quot;sample_day&quot;, Wd1)</code></p>
<pre class="r"><code>physeq &lt;- Wd1

c1 &lt;- c(&quot;soil_type&quot;)
c2 &lt;- c(&quot;sample_day&quot;)

sample_data &lt;- sample_data(physeq) %&gt;%
as_tibble()</code></pre>
<pre><code>## Warning in class(x) &lt;- c(subclass, tibble_class): Setting class(x) to
## multiple strings (&quot;tbl_df&quot;, &quot;tbl&quot;, ...); result will no longer be an S4
## object</code></pre>
<pre class="r"><code>sample_data$alpha &lt;- estimate_richness(physeq, measures = c(&quot;ACE&quot;)) %&gt;% 
  select(ACE) %&gt;%
  unlist()
# Do ANOVA
sample_data &lt;- sample_data %&gt;%
  mutate(grouping = paste(sample_day, soil_type,  sep = &#39;x&#39;))
# needed for how `aov` works
anova_result &lt;- aov(alpha ~ grouping, sample_data)
anova_result</code></pre>
<pre><code>## Call:
##    aov(formula = alpha ~ grouping, data = sample_data)
## 
## Terms:
##                 grouping Residuals
## Sum of Squares   5903523   6282616
## Deg. of Freedom       11       126
## 
## Residual standard error: 223.2981
## Estimated effects may be unbalanced</code></pre>
<pre class="r"><code># Do Tukey&#39;s HSD test
tukey_result &lt;- HSD.test(anova_result, &quot;grouping&quot;, group = TRUE)
# Plot result
group_data &lt;- tukey_result$groups[order(rownames(tukey_result$groups)),]
my_plot &lt;- ggplot(sample_data, aes(x = sample_day, y = alpha)) +
  # geom_text(data = data.frame(),
  #           aes(x = rownames(group_data),
  #               y = max(sample_data$alpha) + 1,
  #               label = group_data$groups),
  #           col = &#39;black&#39;,
  #           size = 10) +
  geom_boxplot(aes(color = soil_type), position = &quot;dodge&quot;) +
  ggtitle(&quot;Alpha diversity&quot;) +
  ylab(&quot;Alpha diversity index&quot;)
my_plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-2-1.png" width="816" /></p>
</div>
<div id="ggpubr-makes-stats-a-lot-easier" class="section level1">
<h1>ggpubr makes stats a lot easier</h1>
<pre class="r"><code>library(ggpubr)
sample_data &lt;- sample_data(Wd1) %&gt;%
  lapply(., factor) %&gt;%
  as_tibble() 
  
richness &lt;- estimate_richness(Wd1) 
sample_data &lt;- cbind(sample_data, richness)

p &lt;- ggboxplot(sample_data, x = &quot;sample_day&quot;, y = &quot;ACE&quot;, fill = &quot;soil_type&quot;, add = &quot;jitter&quot;)

plot &lt;- p + stat_compare_means(aes(group = soil_type), label = &quot;p.signif&quot;) +
  ggtitle(&quot;ACE diversity index of WORLE detph 1 soils&quot;) +
  ylab(&quot;ACE Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + scale_fill_viridis_d(name = &quot;Managment&quot;, labels = c(&quot;Ag&quot;, &quot;Prairie&quot;))
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/ACE-1.png" width="816" /></p>
<pre class="r"><code>ggsave(&quot;images/WorleD1ACE.png&quot;, plot = plot, device = &quot;png&quot;, width = 6, height = 4, units = &quot;in&quot;)</code></pre>
<p>We see some differences on T002, which was two days after the rainfall simulation. The added moisture likely caused an increase in bacteria. Had we measured microbial biomass carbon, I would have expected to see an increase. We could confirm the increase by using 16s primers and qPCR for these samples.</p>
<pre class="r"><code>library(ggpubr)
sample_data &lt;- sample_data(Wd1) %&gt;%
  lapply(., factor) %&gt;%
  as_tibble() 
  
richness &lt;- estimate_richness(Wd1) 
sample_data &lt;- cbind(sample_data, richness)

p &lt;- ggboxplot(sample_data, x = &quot;sample_day&quot;, y = &quot;Chao1&quot;, fill = &quot;soil_type&quot;, add = &quot;jitter&quot;)

plot &lt;- p + stat_compare_means(aes(group = soil_type), label = &quot;p.signif&quot;) +
  ggtitle(&quot;Chao1 diversity index of WORLE detph 1 soils&quot;) +
  ylab(&quot;Chao1 Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + scale_fill_viridis_d(name = &quot;Managment&quot;, labels = c(&quot;Ag&quot;, &quot;Prairie&quot;))
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/Chao1-1.png" width="816" /></p>
<pre class="r"><code>ggsave(&quot;images/WorleD1Chao1.png&quot;, plot = plot, device = &quot;png&quot;, width = 6, height = 4, units = &quot;in&quot;)</code></pre>
<pre class="r"><code>library(ggpubr)
sample_data &lt;- sample_data(Wd1) %&gt;%
  lapply(., factor) %&gt;%
  as_tibble() 
  
richness &lt;- estimate_richness(Wd1) 
sample_data &lt;- cbind(sample_data, richness)

p &lt;- ggboxplot(sample_data, x = &quot;sample_day&quot;, y = &quot;Shannon&quot;, fill = &quot;soil_type&quot;, add = &quot;jitter&quot;)

plot &lt;- p + stat_compare_means(aes(group = soil_type), label = &quot;p.signif&quot;) +
  ggtitle(&quot;Shannon diversity index of WORLE detph 1 soils&quot;) +
  ylab(&quot;Shannon Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + scale_fill_viridis_d(name = &quot;Managment&quot;, labels = c(&quot;Ag&quot;, &quot;Prairie&quot;))
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/Shannon-1.png" width="816" /></p>
<pre class="r"><code>ggsave(&quot;images/WorleD1Shannon.png&quot;, plot = plot, device = &quot;png&quot;, width = 6, height = 4, units = &quot;in&quot;)</code></pre>
<pre class="r"><code>rWd1 &lt;- rarefy_even_depth(Wd1, sample.size = 10000, rngseed = 12121212, trimOTUs = T)</code></pre>
<pre><code>## `set.seed(12121212)` was used to initialize repeatable random subsampling.</code></pre>
<pre><code>## Please record this for your records so others can reproduce.</code></pre>
<pre><code>## Try `set.seed(12121212); .Random.seed` for the full vector</code></pre>
<pre><code>## ...</code></pre>
<pre><code>## 14 samples removedbecause they contained fewer reads than `sample.size`.</code></pre>
<pre><code>## Up to first five removed samples are:</code></pre>
<pre><code>## P1-s3-d1-bP1-s3-d1-t0P1-s6-d1-bP1-s8-d1-t21P1-s9-d1-b</code></pre>
<pre><code>## ...</code></pre>
<pre><code>## 2313OTUs were removed because they are no longer 
## present in any sample after random subsampling</code></pre>
<pre><code>## ...</code></pre>
<p>Rarefied Diversity</p>
<p>Rarefaction is important for making sound judgements when comparing two sites (samples) in this case, our sequencing wasn’t even between all samples, so we are randomly re-sampling each at 10k reads. This removes a few samples that had less than 10k reads, we cannot include them in the comparison becuase the were undersampled. Yet, our conclusions remain the same (T002 differences).</p>
<pre class="r"><code>library(ggpubr)
sample_data &lt;- sample_data(rWd1) %&gt;%
  lapply(., factor) %&gt;%
as_tibble()
richness &lt;- estimate_richness(rWd1) 
sample_data &lt;- cbind(sample_data, richness)
colnames(sample_data)</code></pre>
<pre><code>##  [1] &quot;unique_id&quot;        &quot;experiment&quot;       &quot;matrix&quot;          
##  [4] &quot;treatment&quot;        &quot;plot&quot;             &quot;sample_day&quot;      
##  [7] &quot;depth&quot;            &quot;in_plot_location&quot; &quot;block&quot;           
## [10] &quot;strip&quot;            &quot;manure_treatment&quot; &quot;soil_type&quot;       
## [13] &quot;Observed&quot;         &quot;Chao1&quot;            &quot;se.chao1&quot;        
## [16] &quot;ACE&quot;              &quot;se.ACE&quot;           &quot;Shannon&quot;         
## [19] &quot;Simpson&quot;          &quot;InvSimpson&quot;       &quot;Fisher&quot;</code></pre>
<pre class="r"><code>p &lt;- ggboxplot(sample_data, x = &quot;sample_day&quot;, y = &quot;ACE&quot;, fill = &quot;soil_type&quot;, add = &quot;jitter&quot;)

plot &lt;- p + stat_compare_means(aes(group = soil_type), label = &quot;p.signif&quot;) +
  ggtitle(&quot;ACE diversity index of WORLE detph 1 soils, rarefied to 10k&quot;) +
  ylab(&quot;ACE Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + scale_fill_viridis_d(name = &quot;Managment&quot;, labels = c(&quot;Ag&quot;, &quot;Prairie&quot;))
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/ACE%20rarefied-1.png" width="816" /></p>
<pre class="r"><code>ggsave(&quot;images/rWd1ACE.png&quot;, plot = plot, device = &quot;png&quot;, width = 6, height = 4, units = &quot;in&quot;)</code></pre>
<pre class="r"><code>p &lt;- ggboxplot(sample_data, x = &quot;sample_day&quot;, y = &quot;Chao1&quot;, fill = &quot;soil_type&quot;, add = &quot;jitter&quot;)

plot &lt;- p + stat_compare_means(aes(group = soil_type), label = &quot;p.signif&quot;) +
  ggtitle(&quot;Chao1 diversity index of WORLE detph 1 soils, rarefied to 10k&quot;) +
  ylab(&quot;Chao1 Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + scale_fill_viridis_d(name = &quot;Managment&quot;, labels = c(&quot;Ag&quot;, &quot;Prairie&quot;))
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/Chao1%20rarefied-1.png" width="816" /></p>
<pre class="r"><code>ggsave(&quot;images/rWd1Chao1.png&quot;, plot = plot, device = &quot;png&quot;, width = 6, height = 4, units = &quot;in&quot;)</code></pre>
<pre class="r"><code>p &lt;- ggboxplot(sample_data, x = &quot;sample_day&quot;, y = &quot;Shannon&quot;, fill = &quot;soil_type&quot;, add = &quot;jitter&quot;)

plot &lt;- p + stat_compare_means(aes(group = soil_type), label = &quot;p.signif&quot;) +
  ggtitle(&quot;Shannon diversity index of WORLE detph 1 soils, rarefied to 10k&quot;) +
  ylab(&quot;Shannon Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + scale_fill_viridis_d(name = &quot;Managment&quot;, labels = c(&quot;Ag&quot;, &quot;Prairie&quot;))
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/Shannon%20rarefied-1.png" width="816" /></p>
<pre class="r"><code>ggsave(&quot;images/rWd1Shannon.png&quot;, plot = plot, device = &quot;png&quot;, width = 6, height = 4, units = &quot;in&quot;)</code></pre>
</div>
<div id="prairie-pairwise" class="section level1">
<h1>Prairie Pairwise</h1>
<p>Just prairie soils pairwise comparison, we can see if any of the days were signigicantly more or less diverse than the other. The lines on the graph conncet the two samples that are being compared.</p>
<pre class="r"><code>levels(as.factor(sample_data$sample_day))</code></pre>
<pre><code>## [1] &quot;Baseline&quot; &quot;T000&quot;     &quot;T002&quot;     &quot;T014&quot;     &quot;T021&quot;     &quot;T042&quot;</code></pre>
<pre class="r"><code>levels(as.factor(p$sample_day))</code></pre>
<pre><code>## character(0)</code></pre>
<pre class="r"><code>p &lt;- sample_data %&gt;%
  filter(soil_type == &quot;strip&quot;) %&gt;%
  ggboxplot(x = &quot;sample_day&quot;, y = &quot;Shannon&quot;, add = &quot;jitter&quot;)
p</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-4-1.png" width="816" /></p>
<pre class="r"><code>my_comparisons &lt;- list( c(&quot;T002&quot;, &quot;T014&quot;), c(&quot;T002&quot;, &quot;T000&quot;), c(&quot;Baseline&quot;, &quot;T000&quot;), c(&quot;T014&quot;, &quot;T021&quot;), c(&quot;T021&quot;, &quot;T042&quot;))

plot &lt;- p + stat_compare_means(comparisons = my_comparisons, label = &quot;p.signif&quot;) +
  ggtitle(&quot;Shannon diversity index of WORLE detph 1 Prairie soils, rarefied to 10k&quot;) +
  ylab(&quot;Shannon Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) 
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-4-2.png" width="816" /></p>
<pre class="r"><code>levels(as.factor(sample_data$sample_day))</code></pre>
<pre><code>## [1] &quot;Baseline&quot; &quot;T000&quot;     &quot;T002&quot;     &quot;T014&quot;     &quot;T021&quot;     &quot;T042&quot;</code></pre>
<pre class="r"><code>levels(as.factor(p$sample_day))</code></pre>
<pre><code>## character(0)</code></pre>
<pre class="r"><code>p &lt;- sample_data %&gt;%
  filter(soil_type == &quot;strip&quot;) %&gt;%
  ggboxplot(x = &quot;sample_day&quot;, y = &quot;ACE&quot;, add = &quot;jitter&quot;)
p</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-5-1.png" width="816" /></p>
<pre class="r"><code>my_comparisons &lt;- list( c(&quot;T002&quot;, &quot;T014&quot;), c(&quot;T002&quot;, &quot;T000&quot;), c(&quot;Baseline&quot;, &quot;T000&quot;), c(&quot;T014&quot;, &quot;T021&quot;), c(&quot;T021&quot;, &quot;T042&quot;))

plot &lt;- p + stat_compare_means(comparisons = my_comparisons, label = &quot;p.signif&quot;) +
  ggtitle(&quot;ACE diversity index of WORLE depth 1 Prairie soils, rarefied to 10k&quot;) +
  ylab(&quot;ACE Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) 
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-5-2.png" width="816" /></p>
<p>Trying a violin plot</p>
<pre class="r"><code>p &lt;- sample_data %&gt;%
  filter(soil_type == &quot;crop&quot;) %&gt;%
  ggviolin(x = &quot;sample_day&quot;, y = &quot;ACE&quot;, fill = &quot;sample_day&quot;,
         add = &quot;boxplot&quot;, add.params = list(fill = &quot;gray&quot;))
p</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-6-1.png" width="816" /></p>
<pre class="r"><code>my_comparisons &lt;- list( c(&quot;T002&quot;, &quot;T014&quot;), c(&quot;T002&quot;, &quot;T000&quot;), c(&quot;Baseline&quot;, &quot;T000&quot;), c(&quot;T014&quot;, &quot;T021&quot;), c(&quot;T021&quot;, &quot;T042&quot;))

plot &lt;- p + stat_compare_means(comparisons = my_comparisons, label = &quot;p.signif&quot;) +
  ggtitle(&quot;ACE diversity index of WORLE depth 1 Crop soils, rarefied to 10k&quot;) +
  ylab(&quot;ACE Diversity&quot;) +
  xlab(&quot;Sampling Day&quot;) + 
  scale_fill_viridis_d() + 
  theme(legend.position = &quot;none&quot;) 
plot</code></pre>
<p><img src="Diversity_Stats_files/figure-html/unnamed-chunk-6-2.png" width="816" /></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
